gobblin-aws/src/main/java/gobblin/aws/AWSJobConfigurationManager.java

```diff
@@ -0,0 +1,230 @@
+import gobblin.cluster.JobConfigurationManager;
+import gobblin.cluster.event.NewJobConfigArrivalEvent;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.util.ExecutorsUtils;
+import gobblin.util.SchedulerUtils;
+import static gobblin.aws.GobblinAWSUtils.*;
+
+
+/**
+ * Class for managing AWS Gobblin job configurations.
+ *
+ * <p>
+ *   This class reads all the job configuration at startup from S3
+ *   and schedules a refresh to poll from S3 for any new job configurations.
+ *   The jobs read are scheduled by the {@link GobblinHelixJobScheduler} by posting a
+ *   {@link NewJobConfigArrivalEvent} for each job configuration file.
+ * </p>
+ *
+ * @author Abhishek Tiwari
+ */
+@Alpha
+public class AWSJobConfigurationManager extends JobConfigurationManager {
+  private static final Logger LOGGER = LoggerFactory.getLogger(AWSJobConfigurationManager.class);
+
+  private Optional<String> jobConfS3Uri;
+  private Map<String, Properties> jobConfFiles;
+
+  private final long refreshIntervalInSeconds;
+
+  private final ScheduledExecutorService fetchJobConfExecutor;
+
+  public AWSJobConfigurationManager(EventBus eventBus, Config config) {
+    super(eventBus, config);
+    this.jobConfFiles = Maps.newHashMap();
+    if (config.hasPath(GobblinAWSConfigurationKeys.JOB_CONF_REFRESH_INTERVAL_IN_SECONDS)) {
+      this.refreshIntervalInSeconds = config
+          .getLong(GobblinAWSConfigurationKeys.JOB_CONF_REFRESH_INTERVAL_IN_SECONDS);
+    } else {
+      this.refreshIntervalInSeconds = config
+          .getLong(GobblinAWSConfigurationKeys.JOB_CONF_REFRESH_INTERVAL_IN_MINUTES) * 60;
+    }
+
+    this.fetchJobConfExecutor = Executors.newSingleThreadScheduledExecutor(
+        ExecutorsUtils.newThreadFactory(Optional.of(LOGGER), Optional.of("FetchJobConfExecutor")));
+  }
+
+  private void fetchJobConfSettings() {
+    this.jobConfDirPath =
+        config.hasPath(GobblinClusterConfigurationKeys.JOB_CONF_PATH_KEY) ? Optional
+            .of(config.getString(GobblinClusterConfigurationKeys.JOB_CONF_PATH_KEY)) : Optional.<String>absent();
+    this.jobConfS3Uri =
+        config.hasPath(GobblinAWSConfigurationKeys.JOB_CONF_S3_URI_KEY) ? Optional
+            .of(config.getString(GobblinAWSConfigurationKeys.JOB_CONF_S3_URI_KEY)) : Optional.<String>absent();
+  }
+
+  @Override
+  protected void startUp() throws Exception {
+    LOGGER.info("Starting the " + AWSJobConfigurationManager.class.getSimpleName());
+
+    LOGGER.info(String.format("Scheduling the job configuration refresh task with an interval of %d second(s)",
+            this.refreshIntervalInSeconds));
+
+    // Schedule the job config fetch task
+    this.fetchJobConfExecutor.scheduleAtFixedRate(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          fetchJobConf();
+        } catch (IOException | ConfigurationException e) {
+          LOGGER.error("Failed to fetch job configurations", e);
+          throw Throwables.propagate(e);
```

Comment:
It was remanent of pre-refactor code. However, it has some advantages in some situations: https://github.com/google/guava/wiki/ThrowablesExplained 

However, this doesn't needs it. Changed